src/ashell/ashell.c

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) 2016-2017, Intel Corporation.
+
+/**
+ * @file
+ * @brief Ashell main.
+ */
+
+#include "ashell.h"
+
+void ide_init()
+{
+    extern void parser_init();
```

Comment:
I personally would prefer to have header files to #include vs using extern.  Is there a benefit to this method that I'm not aware of?

Reply:
Will do this a bit later, but before merging (when removing the WiP flag). Then we'll include what is strictly needed.